unit-testing/src/mock-function/lib/insert.ts

```diff
@@ -0,0 +1,41 @@
+// insertDocument.ts
+import { Container } from '../data/connect-to-cosmos';
+import {
+  DbDocument,
+  DbError,
+  RawInput,
+  VerificationErrors,
+} from '../data/model-schema';
+import { inputVerified } from '../data/verify';
+
+export async function insertDocument(
+  container: Container,
+  doc: RawInput,
+): Promise<DbDocument | DbError | VerificationErrors> {
+  const isVerified: boolean = inputVerified(doc);
+
+  if (!isVerified) {
+    return { message: 'Verification failed' } as VerificationErrors;
+  }
+
+  try {
+    const { resource } = await container.items.create({
+      id: doc.id,
+      name: `${doc.first} ${doc.last}`,
+    });
+
+    return resource as DbDocument;
+  } catch (error: unknown) {
```

Comment:
I usually use `any` for errors - it's one of the only places where I still use `any` instead of `unknown`

While it's true that technically _anything_ can be thrown (`throw "foo"` for example) in there is nothing here that would fail unless you `throw null` or `throw undefined` - all primitives (`throw "foo"`, `throw 5`, etc) and objects will just return undefined when you grab their properties (like `code`, etc) so for simple error handling I prefer to use `any` here. Happy to defer to you, if you would rather stick to `unknown` for any reason

Reply:
Any works